src/main/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/service/DelegatedAccessService.kt

```diff
@@ -20,30 +20,25 @@ class DelegatedAccessService(private val delegatedAccessRepository: DelegatedAcc
   fun createDelegatedAccess(delegatePost: DelegatedAccess): DelegatedAccessEntity {
     val now = LocalDateTime.now()
 
-    val initiatedUserExists = userRepository.existsById(delegatePost.initiatedUserId.toLong()) ?: throw ResourceNotFoundException("User with id ${delegatePost.initiatedUserId} not found in database ")
-    val delegatedUserExists = userRepository.existsById(delegatePost.delegatedUserId.toLong()) ?: throw ResourceNotFoundException("User with id ${delegatePost.delegatedUserId} not found in database ")
-
-    if (initiatedUserExists && delegatedUserExists) {
-      val delegatedAccessAlreadyExists = delegatedAccessRepository.findByInitiatedUserIdAndDelegatedUserIdAndDeletedDateIsNull(delegatePost.initiatedUserId.toLong(), delegatePost.delegatedUserId.toLong())
-      if (delegatedAccessAlreadyExists != null) {
-        throw DuplicateDataFoundException("Delegated Access already exists and active!")
-      }
-      val delegatedAccessEntity = DelegatedAccessEntity(
-        id = null,
-        initiatedUserId = delegatePost.initiatedUserId,
-        delegatedUserId = delegatePost.delegatedUserId,
-        createdDate = now,
-        deletedDate = null,
-      )
-      return delegatedAccessRepository.save(delegatedAccessEntity)
-    } else {
-      throw ValidationException(
-        "Request invalid. " +
-          "initiatedUserId= ${delegatePost.initiatedUserId} " +
-          "delegatedUserId= ${delegatePost.delegatedUserId} ",
+    val initiatedUser = userRepository.findById(delegatePost.initiatedUserId.toLong()) ?: throw ResourceNotFoundException("User with id ${delegatePost.initiatedUserId} not found in database ")
+    val delegatedUser = userRepository.findById(delegatePost.delegatedUserId.toLong()) ?: throw ResourceNotFoundException("User with id ${delegatePost.delegatedUserId} not found in database ")
 
+    val delegatedAccessAlreadyExists =
+      delegatedAccessRepository.findByInitiatedUserIdAndDelegatedUserIdAndDeletedDateIsNull(
+        initiatedUser.get().id!!.toLong(),
+        delegatedUser.get().id!!.toLong(),
```

Comment:
we should declare the input as a long, else we may end up with unsubmittable user ids

Reply:
Done